packages/cbioportal-utils/src/civic/CivicUtils.ts

```diff
@@ -24,42 +24,15 @@ export enum CivicAlterationType {
 
 export const CIVIC_NA_VALUE = 'NA';
 
-/**
- * Asynchronously adds the given variant from a gene to the variant map specified.
- */
-function lookupCivicVariantAndAddToMap(
-    variantMap: ICivicVariantIndex,
-    variantId: number,
-    variantName: string,
-    geneSymbol: string,
-    geneId: number
-): Promise<void> {
-    return civicClient
-        .getCivicVariantSummary(variantId, variantName, geneId)
-        .then((result: ICivicVariantSummary) => {
-            if (result) {
-                if (!variantMap[geneSymbol]) {
-                    variantMap[geneSymbol] = {};
-                }
-                variantMap[geneSymbol][variantName] = result;
-            }
-        });
-}
-
 /**
  * Asynchronously return a map with Civic information from the genes given.
  */
-export function getCivicGenes(
-    entrezGeneIds: number[]
-): Promise<ICivicGeneIndex> {
+export function getCivicGenes(hugoSymbols: string[]): Promise<ICivicGeneIndex> {
     // Assemble a list of promises, each of which will retrieve a batch of genes
     let promises: Array<Promise<Array<ICivicGeneSummary>>> = [];
 
-    // To prevent the request from growing too large, we send it off in multiple chunks
-    const chunkedIds: number[][] = _.chunk(_.uniq(entrezGeneIds), 400);
-
-    chunkedIds.forEach(entrezGeneIds =>
-        promises.push(civicClient.getCivicGenesBatch(entrezGeneIds.join(',')))
+    hugoSymbols.forEach(hugoSymbol =>
+        promises.push(civicClient.getCivicGenesBatch(hugoSymbol))
```

Comment:
Yes that's the problem right now. I don't see a way to send "POST" queries that can accept a list of hugo gene symbols. As far as I can see it can only accept single gene or giving the "start civic id" and "end civic id", the response will then include all genes between this range, but we still need to send extra queries to get the civic id, so it's still not ideal. 
For results view it only sends one gene, but for copy number alterations on patients view it sends too many requests. I already asked civic team to see if they have a better way to handle it.

Reply:
This is solved. Civic team just made a new release yesterday, they can support a list of hugo symbols now in their genes API.